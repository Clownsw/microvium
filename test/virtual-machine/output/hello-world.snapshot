

slot ['dummy.mvms:#entry'] = &function ['dummy.mvms:#entry'];
slot ['dummy.mvms:exports'] = &allocation 1;
slot ['global:print'] = &ephemeral print;

function ['dummy.mvms:#entry']() {
  entry:
    LoadArg(index 0);
    StoreGlobal(name 'dummy.mvms:exports');
    LoadGlobal(name 'global:print');
    Literal(lit 'Hello, World!');
    Call(count 1);
    Pop(count 1);
    Literal(lit undefined);
    Return();
}

allocation 1 = {
};