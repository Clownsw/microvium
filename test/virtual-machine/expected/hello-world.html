
    <!DOCTYPE html>
    <html lang="en">
    <head>
      <meta charset="UTF-8">
      <meta name="viewport" content="width=device-width, initial-scale=1.0">
      </head>
      <body>
      
  <table class="visual-buffer">
    <colgroup>
      <col>
      <col>
      <col>
      <col>
    </colgroup>
    <tbody>
      
      <tr>
        <td class="address">
          <a class="address-text" id="address0000" href="#address0000">
            0000
          </a>
        </td>
        <td class="data">
          <span class="byte">01</span>
        </td>
        <td class="label">
          bytecodeVersion: 
        </td>
        <td class="value">1</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address0001" href="#address0001">
            0001
          </a>
        </td>
        <td class="data">
          <span class="byte">2C</span>
        </td>
        <td class="label">
          headerSize: 
        </td>
        <td class="value">44</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address0002" href="#address0002">
            0002
          </a>
        </td>
        <td class="data">
          <span class="byte">B9</span><wbr><span class="byte">00</span>
        </td>
        <td class="label">
          bytecodeSize: 
        </td>
        <td class="value">185</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address0004" href="#address0004">
            0004
          </a>
        </td>
        <td class="data">
          <span class="byte">E0</span><wbr><span class="byte">72</span>
        </td>
        <td class="label">
          crc: 
        </td>
        <td class="value">0x72E0</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address0006" href="#address0006">
            0006
          </a>
        </td>
        <td class="data">
          <span class="byte">00</span><wbr><span class="byte">00</span>
        </td>
        <td class="label">
          requiredEngineVersion: 
        </td>
        <td class="value">0</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address0008" href="#address0008">
            0008
          </a>
        </td>
        <td class="data">
          <span class="byte">00</span><wbr><span class="byte">00</span><wbr><span class="byte">00</span><wbr><span class="byte">00</span>
        </td>
        <td class="label">
          requiredFeatureFlags: 
        </td>
        <td class="value">0x00000000</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address000C" href="#address000C">
            000C
          </a>
        </td>
        <td class="data">
          <span class="byte">0A</span><wbr><span class="byte">00</span>
        </td>
        <td class="label">
          globalVariableCount: 
        </td>
        <td class="value">10</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address000E" href="#address000E">
            000E
          </a>
        </td>
        <td class="data">
          <span class="byte">2C</span><wbr><span class="byte">00</span>
        </td>
        <td class="label">
          initialDataOffset: 
        </td>
        <td class="value">0x002C</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address0010" href="#address0010">
            0010
          </a>
        </td>
        <td class="data">
          <span class="byte">14</span><wbr><span class="byte">00</span>
        </td>
        <td class="label">
          initialDataSize: 
        </td>
        <td class="value">20</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address0012" href="#address0012">
            0012
          </a>
        </td>
        <td class="data">
          <span class="byte">40</span><wbr><span class="byte">00</span>
        </td>
        <td class="label">
          initialHeapOffset: 
        </td>
        <td class="value">0x0040</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address0014" href="#address0014">
            0014
          </a>
        </td>
        <td class="data">
          <span class="byte">18</span><wbr><span class="byte">00</span>
        </td>
        <td class="label">
          initialHeapSize: 
        </td>
        <td class="value">24</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address0016" href="#address0016">
            0016
          </a>
        </td>
        <td class="data">
          <span class="byte">58</span><wbr><span class="byte">00</span>
        </td>
        <td class="label">
          gcRootsOffset: 
        </td>
        <td class="value">0x0058</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address0018" href="#address0018">
            0018
          </a>
        </td>
        <td class="data">
          <span class="byte">00</span><wbr><span class="byte">00</span>
        </td>
        <td class="label">
          gcRootsCount: 
        </td>
        <td class="value">0</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address001A" href="#address001A">
            001A
          </a>
        </td>
        <td class="data">
          <span class="byte">58</span><wbr><span class="byte">00</span>
        </td>
        <td class="label">
          importTableOffset: 
        </td>
        <td class="value">0x0058</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address001C" href="#address001C">
            001C
          </a>
        </td>
        <td class="data">
          <span class="byte">00</span><wbr><span class="byte">00</span>
        </td>
        <td class="label">
          importTableSize: 
        </td>
        <td class="value">0</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address001E" href="#address001E">
            001E
          </a>
        </td>
        <td class="data">
          <span class="byte">58</span><wbr><span class="byte">00</span>
        </td>
        <td class="label">
          exportTableOffset: 
        </td>
        <td class="value">0x0058</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address0020" href="#address0020">
            0020
          </a>
        </td>
        <td class="data">
          <span class="byte">00</span><wbr><span class="byte">00</span>
        </td>
        <td class="label">
          exportTableSize: 
        </td>
        <td class="value">0</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address0022" href="#address0022">
            0022
          </a>
        </td>
        <td class="data">
          <span class="byte">58</span><wbr><span class="byte">00</span>
        </td>
        <td class="label">
          shortCallTableOffset: 
        </td>
        <td class="value">0x0058</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address0024" href="#address0024">
            0024
          </a>
        </td>
        <td class="data">
          <span class="byte">00</span><wbr><span class="byte">00</span>
        </td>
        <td class="label">
          shortCallTableSize: 
        </td>
        <td class="value">0</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address0026" href="#address0026">
            0026
          </a>
        </td>
        <td class="data">
          <span class="byte">58</span><wbr><span class="byte">00</span>
        </td>
        <td class="label">
          stringTableOffset: 
        </td>
        <td class="value">0x0058</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address0028" href="#address0028">
            0028
          </a>
        </td>
        <td class="data">
          <span class="byte">04</span><wbr><span class="byte">00</span>
        </td>
        <td class="label">
          stringTableSize: 
        </td>
        <td class="value">4</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address002A" href="#address002A">
            002A
          </a>
        </td>
        <td class="data">
          <span class="byte">00</span><wbr><span class="byte">00</span>
        </td>
        <td class="label">
          reserved: 
        </td>
        <td class="value">0</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address002C" href="#address002C">
            002C
          </a>
        </td>
        <td class="data">
          <span class="byte">02</span><wbr><span class="byte">40</span>
        </td>
        <td class="label">
          <builtin>:exports: 
        </td>
        <td class="value">0x4002</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address002E" href="#address002E">
            002E
          </a>
        </td>
        <td class="data">
          <span class="byte">8F</span><wbr><span class="byte">C0</span>
        </td>
        <td class="label">
          <builtin>:#entry: 
        </td>
        <td class="value">0xC08F</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address0030" href="#address0030">
            0030
          </a>
        </td>
        <td class="data">
          <span class="byte">9C</span><wbr><span class="byte">C0</span>
        </td>
        <td class="label">
          <builtin>:Number_isNaN: 
        </td>
        <td class="value">0xC09C</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address0032" href="#address0032">
            0032
          </a>
        </td>
        <td class="data">
          <span class="byte">5E</span><wbr><span class="byte">C0</span>
        </td>
        <td class="label">
          global:Infinity: 
        </td>
        <td class="value">0xC05E</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address0034" href="#address0034">
            0034
          </a>
        </td>
        <td class="data">
          <span class="byte">15</span><wbr><span class="byte">C0</span>
        </td>
        <td class="label">
          global:NaN: 
        </td>
        <td class="value">0xC015</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address0036" href="#address0036">
            0036
          </a>
        </td>
        <td class="data">
          <span class="byte">11</span><wbr><span class="byte">C0</span>
        </td>
        <td class="label">
          global:undefined: 
        </td>
        <td class="value">0xC011</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address0038" href="#address0038">
            0038
          </a>
        </td>
        <td class="data">
          <span class="byte">0C</span><wbr><span class="byte">40</span>
        </td>
        <td class="label">
          global:Number: 
        </td>
        <td class="value">0x400C</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address003A" href="#address003A">
            003A
          </a>
        </td>
        <td class="data">
          <span class="byte">B3</span><wbr><span class="byte">C0</span>
        </td>
        <td class="label">
          global:print: 
        </td>
        <td class="value">0xC0B3</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address003C" href="#address003C">
            003C
          </a>
        </td>
        <td class="data">
          <span class="byte">16</span><wbr><span class="byte">40</span>
        </td>
        <td class="label">
          dummy.mvms:exports: 
        </td>
        <td class="value">0x4016</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address003E" href="#address003E">
            003E
          </a>
        </td>
        <td class="data">
          <span class="byte">A6</span><wbr><span class="byte">C0</span>
        </td>
        <td class="label">
          dummy.mvms:#entry: 
        </td>
        <td class="value">0xC0A6</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address0040" href="#address0040">
            0040
          </a>
        </td>
        <td class="data">
          <span class="byte">01</span><wbr><span class="byte">50</span>
        </td>
        <td class="label">
          TsPropertyList.[header]: 
        </td>
        <td class="value">0x5001</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address0042" href="#address0042">
            0042
          </a>
        </td>
        <td class="data">
          <span class="byte">04</span><wbr><span class="byte">40</span>
        </td>
        <td class="label">
          TsPropertyList.[first]: 
        </td>
        <td class="value">0x4004</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address0044" href="#address0044">
            0044
          </a>
        </td>
        <td class="data">
          <span class="byte">00</span><wbr><span class="byte">00</span>
        </td>
        <td class="label">
          TsPropertyCell.[next]: 
        </td>
        <td class="value">0x0000</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address0046" href="#address0046">
            0046
          </a>
        </td>
        <td class="data">
          <span class="byte">68</span><wbr><span class="byte">C0</span>
        </td>
        <td class="label">
          TsPropertyCell.[key]: 
        </td>
        <td class="value">0xC068</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address0048" href="#address0048">
            0048
          </a>
        </td>
        <td class="data">
          <span class="byte">9C</span><wbr><span class="byte">C0</span>
        </td>
        <td class="label">
          Object.Number_isNaN: 
        </td>
        <td class="value">0xC09C</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address004A" href="#address004A">
            004A
          </a>
        </td>
        <td class="data">
          <span class="byte">01</span><wbr><span class="byte">50</span>
        </td>
        <td class="label">
          TsPropertyList.[header]: 
        </td>
        <td class="value">0x5001</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address004C" href="#address004C">
            004C
          </a>
        </td>
        <td class="data">
          <span class="byte">0E</span><wbr><span class="byte">40</span>
        </td>
        <td class="label">
          TsPropertyList.[first]: 
        </td>
        <td class="value">0x400E</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address004E" href="#address004E">
            004E
          </a>
        </td>
        <td class="data">
          <span class="byte">00</span><wbr><span class="byte">00</span>
        </td>
        <td class="label">
          TsPropertyCell.[next]: 
        </td>
        <td class="value">0x0000</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address0050" href="#address0050">
            0050
          </a>
        </td>
        <td class="data">
          <span class="byte">77</span><wbr><span class="byte">C0</span>
        </td>
        <td class="label">
          TsPropertyCell.[key]: 
        </td>
        <td class="value">0xC077</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address0052" href="#address0052">
            0052
          </a>
        </td>
        <td class="data">
          <span class="byte">9C</span><wbr><span class="byte">C0</span>
        </td>
        <td class="label">
          Object.isNaN: 
        </td>
        <td class="value">0xC09C</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address0054" href="#address0054">
            0054
          </a>
        </td>
        <td class="data">
          <span class="byte">00</span><wbr><span class="byte">50</span>
        </td>
        <td class="label">
          TsPropertyList.[header]: 
        </td>
        <td class="value">0x5000</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address0056" href="#address0056">
            0056
          </a>
        </td>
        <td class="data">
          <span class="byte">00</span><wbr><span class="byte">00</span>
        </td>
        <td class="label">
          TsPropertyList.[first]: 
        </td>
        <td class="value">0x0000</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address0058" href="#address0058">
            0058
          </a>
        </td>
        <td class="data">
          <span class="byte">68</span><wbr><span class="byte">C0</span>
        </td>
        <td class="label">
          &Number_isNaN: 
        </td>
        <td class="value">0xC068</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address005A" href="#address005A">
            005A
          </a>
        </td>
        <td class="data">
          <span class="byte">77</span><wbr><span class="byte">C0</span>
        </td>
        <td class="label">
          &isNaN: 
        </td>
        <td class="value">0xC077</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address005C" href="#address005C">
            005C
          </a>
        </td>
        <td class="data">
          <span class="byte">08</span><wbr><span class="byte">20</span>
        </td>
        <td class="label">
          Allocation header: 
        </td>
        <td class="value">0x2008</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address005E" href="#address005E">
            005E
          </a>
        </td>
        <td class="data">
          <span class="byte">00</span><wbr><span class="byte">00</span><wbr><span class="byte">00</span><wbr><span class="byte">00</span><wbr><span class="byte">00</span><wbr><span class="byte">00</span><wbr><span class="byte">F0</span><wbr><span class="byte">7F</span>
        </td>
        <td class="label">
          Double: 
        </td>
        <td class="value">Infinity</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address0066" href="#address0066">
            0066
          </a>
        </td>
        <td class="data">
          <span class="byte">0D</span><wbr><span class="byte">40</span>
        </td>
        <td class="label">
          Allocation header: 
        </td>
        <td class="value">0x400D</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address0068" href="#address0068">
            0068
          </a>
        </td>
        <td class="data">
          <span class="byte">4E</span><wbr><span class="byte">75</span><wbr><span class="byte">6D</span><wbr><span class="byte">62</span><wbr><span class="byte">65</span><wbr><span class="byte">72</span><wbr><span class="byte">5F</span><wbr><span class="byte">69</span><wbr><span class="byte">73</span><wbr><span class="byte">4E</span><wbr><span class="byte">61</span><wbr><span class="byte">4E</span><wbr><span class="byte">00</span>
        </td>
        <td class="label">
          String: 
        </td>
        <td class="value">&#39;Number_isNaN&#39;</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address0075" href="#address0075">
            0075
          </a>
        </td>
        <td class="data">
          <span class="byte">06</span><wbr><span class="byte">40</span>
        </td>
        <td class="label">
          Allocation header: 
        </td>
        <td class="value">0x4006</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address0077" href="#address0077">
            0077
          </a>
        </td>
        <td class="data">
          <span class="byte">69</span><wbr><span class="byte">73</span><wbr><span class="byte">4E</span><wbr><span class="byte">61</span><wbr><span class="byte">4E</span><wbr><span class="byte">00</span>
        </td>
        <td class="label">
          String: 
        </td>
        <td class="value">&#39;isNaN&#39;</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address007D" href="#address007D">
            007D
          </a>
        </td>
        <td class="data">
          <span class="byte">0E</span><wbr><span class="byte">40</span>
        </td>
        <td class="label">
          Allocation header: 
        </td>
        <td class="value">0x400E</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address007F" href="#address007F">
            007F
          </a>
        </td>
        <td class="data">
          <span class="byte">48</span><wbr><span class="byte">65</span><wbr><span class="byte">6C</span><wbr><span class="byte">6C</span><wbr><span class="byte">6F</span><wbr><span class="byte">2C</span><wbr><span class="byte">20</span><wbr><span class="byte">57</span><wbr><span class="byte">6F</span><wbr><span class="byte">72</span><wbr><span class="byte">6C</span><wbr><span class="byte">64</span><wbr><span class="byte">21</span><wbr><span class="byte">00</span>
        </td>
        <td class="label">
          String: 
        </td>
        <td class="value">&#39;Hello, World!&#39;</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address008D" href="#address008D">
            008D
          </a>
        </td>
        <td class="data">
          <span class="byte">0D</span><wbr><span class="byte">80</span>
        </td>
        <td class="label">
          Func alloc header: 
        </td>
        <td class="value">0x800D</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address008F" href="#address008F">
            008F
          </a>
        </td>
        <td class="data">
          <span class="byte">03</span>
        </td>
        <td class="label">
          maxStackDepth: 
        </td>
        <td class="value">3</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address0090" href="#address0090">
            0090
          </a>
        </td>
        <td class="data">
          <span class="byte">30</span>
        </td>
        <td class="label">
          VM_OP_LOAD_ARG_1(0x0): 
        </td>
        <td class="value">LoadArg(index 0)</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address0091" href="#address0091">
            0091
          </a>
        </td>
        <td class="data">
          <span class="byte">A0</span>
        </td>
        <td class="label">
          VM_OP_STORE_GLOBAL_1(0x0): 
        </td>
        <td class="value">StoreGlobal(name &#39;&lt;builtin&gt;:exports&#39;)</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address0092" href="#address0092">
            0092
          </a>
        </td>
        <td class="data">
          <span class="byte">20</span>
        </td>
        <td class="label">
          VM_OP_LOAD_GLOBAL_1(0x0): 
        </td>
        <td class="value">LoadGlobal(name &#39;&lt;builtin&gt;:exports&#39;)</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address0093" href="#address0093">
            0093
          </a>
        </td>
        <td class="data">
          <span class="byte">71</span><wbr><span class="byte">68</span><wbr><span class="byte">C0</span>
        </td>
        <td class="label">
          VM_OP3_LOAD_LITERAL: 
        </td>
        <td class="value">Literal(lit &#39;Number_isNaN&#39;)</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address0096" href="#address0096">
            0096
          </a>
        </td>
        <td class="data">
          <span class="byte">22</span>
        </td>
        <td class="label">
          VM_OP_LOAD_GLOBAL_1(0x2): 
        </td>
        <td class="value">LoadGlobal(name &#39;&lt;builtin&gt;:Number_isNaN&#39;)</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address0097" href="#address0097">
            0097
          </a>
        </td>
        <td class="data">
          <span class="byte">5A</span>
        </td>
        <td class="label">
          VM_OP1_OBJECT_SET_1: 
        </td>
        <td class="value">ObjectSet()</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address0098" href="#address0098">
            0098
          </a>
        </td>
        <td class="data">
          <span class="byte">01</span>
        </td>
        <td class="label">
          VM_OP_LOAD_SMALL_LITERAL(0x1): 
        </td>
        <td class="value">Literal(lit undefined)</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address0099" href="#address0099">
            0099
          </a>
        </td>
        <td class="data">
          <span class="byte">51</span>
        </td>
        <td class="label">
          VM_OP1_RETURN_2: 
        </td>
        <td class="value">Return()</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address009A" href="#address009A">
            009A
          </a>
        </td>
        <td class="data">
          <span class="byte">0A</span><wbr><span class="byte">80</span>
        </td>
        <td class="label">
          Func alloc header: 
        </td>
        <td class="value">0x800A</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address009C" href="#address009C">
            009C
          </a>
        </td>
        <td class="data">
          <span class="byte">05</span>
        </td>
        <td class="label">
          maxStackDepth: 
        </td>
        <td class="value">5</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address009D" href="#address009D">
            009D
          </a>
        </td>
        <td class="data">
          <span class="byte">31</span>
        </td>
        <td class="label">
          VM_OP_LOAD_ARG_1(0x1): 
        </td>
        <td class="value">LoadArg(index 1)</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address009E" href="#address009E">
            009E
          </a>
        </td>
        <td class="data">
          <span class="byte">10</span>
        </td>
        <td class="label">
          VM_OP_LOAD_VAR_1(0x0): 
        </td>
        <td class="value">LoadVar(index 0)</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address009F" href="#address009F">
            009F
          </a>
        </td>
        <td class="data">
          <span class="byte">11</span>
        </td>
        <td class="label">
          VM_OP_LOAD_VAR_1(0x1): 
        </td>
        <td class="value">LoadVar(index 0)</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address00A0" href="#address00A0">
            00A0
          </a>
        </td>
        <td class="data">
          <span class="byte">59</span>
        </td>
        <td class="label">
          VM_OP_EXTENDED_1(0x9): 
        </td>
        <td class="value">BinOp(op &#39;!==&#39;)</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address00A1" href="#address00A1">
            00A1
          </a>
        </td>
        <td class="data">
          <span class="byte">51</span>
        </td>
        <td class="label">
          VM_OP1_RETURN_2: 
        </td>
        <td class="value">Return()</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address00A2" href="#address00A2">
            00A2
          </a>
        </td>
        <td class="data">
          <span class="byte">01</span>
        </td>
        <td class="label">
          VM_OP_LOAD_SMALL_LITERAL(0x1): 
        </td>
        <td class="value">Literal(lit undefined)</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address00A3" href="#address00A3">
            00A3
          </a>
        </td>
        <td class="data">
          <span class="byte">51</span>
        </td>
        <td class="label">
          VM_OP1_RETURN_2: 
        </td>
        <td class="value">Return()</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address00A4" href="#address00A4">
            00A4
          </a>
        </td>
        <td class="data">
          <span class="byte">0F</span><wbr><span class="byte">80</span>
        </td>
        <td class="label">
          Func alloc header: 
        </td>
        <td class="value">0x800F</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address00A6" href="#address00A6">
            00A6
          </a>
        </td>
        <td class="data">
          <span class="byte">03</span>
        </td>
        <td class="label">
          maxStackDepth: 
        </td>
        <td class="value">3</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address00A7" href="#address00A7">
            00A7
          </a>
        </td>
        <td class="data">
          <span class="byte">30</span>
        </td>
        <td class="label">
          VM_OP_LOAD_ARG_1(0x0): 
        </td>
        <td class="value">LoadArg(index 0)</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address00A8" href="#address00A8">
            00A8
          </a>
        </td>
        <td class="data">
          <span class="byte">A8</span>
        </td>
        <td class="label">
          VM_OP_STORE_GLOBAL_1(0x8): 
        </td>
        <td class="value">StoreGlobal(name &#39;dummy.mvms:exports&#39;)</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address00A9" href="#address00A9">
            00A9
          </a>
        </td>
        <td class="data">
          <span class="byte">27</span>
        </td>
        <td class="label">
          VM_OP_LOAD_GLOBAL_1(0x7): 
        </td>
        <td class="value">LoadGlobal(name &#39;global:print&#39;)</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address00AA" href="#address00AA">
            00AA
          </a>
        </td>
        <td class="data">
          <span class="byte">01</span>
        </td>
        <td class="label">
          VM_OP_LOAD_SMALL_LITERAL(0x1): 
        </td>
        <td class="value">Literal(lit undefined)</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address00AB" href="#address00AB">
            00AB
          </a>
        </td>
        <td class="data">
          <span class="byte">71</span><wbr><span class="byte">7F</span><wbr><span class="byte">C0</span>
        </td>
        <td class="label">
          VM_OP3_LOAD_LITERAL: 
        </td>
        <td class="value">Literal(lit &#39;Hello, World!&#39;)</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address00AE" href="#address00AE">
            00AE
          </a>
        </td>
        <td class="data">
          <span class="byte">68</span><wbr><span class="byte">02</span>
        </td>
        <td class="label">
          VM_OP2_CALL_3(0x2): 
        </td>
        <td class="value">Call(count 2)</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address00B0" href="#address00B0">
            00B0
          </a>
        </td>
        <td class="data">
          <span class="byte">80</span>
        </td>
        <td class="label">
          VM_OP_POP(0x0): 
        </td>
        <td class="value">Pop(count 1)</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address00B1" href="#address00B1">
            00B1
          </a>
        </td>
        <td class="data">
          <span class="byte">01</span>
        </td>
        <td class="label">
          VM_OP_LOAD_SMALL_LITERAL(0x1): 
        </td>
        <td class="value">Literal(lit undefined)</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address00B2" href="#address00B2">
            00B2
          </a>
        </td>
        <td class="data">
          <span class="byte">51</span>
        </td>
        <td class="label">
          VM_OP1_RETURN_2: 
        </td>
        <td class="value">Return()</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address00B3" href="#address00B3">
            00B3
          </a>
        </td>
        <td class="data">
          <span class="byte">06</span><wbr><span class="byte">80</span>
        </td>
        <td class="label">
          Func alloc header: 
        </td>
        <td class="value">0x8006</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address00B5" href="#address00B5">
            00B5
          </a>
        </td>
        <td class="data">
          <span class="byte">00</span>
        </td>
        <td class="label">
          maxStackDepth: 
        </td>
        <td class="value">0</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address00B6" href="#address00B6">
            00B6
          </a>
        </td>
        <td class="data">
          <span class="byte">6D</span><wbr><span class="byte">15</span>
        </td>
        <td class="label">
          Detached ephemeral stub: 
        </td>
        <td class="value">VM_OP4_CALL_DETACHED_EPHEMERAL</td>
      </tr>

      <tr>
        <td class="address">
          <a class="address-text" id="address00B8" href="#address00B8">
            00B8
          </a>
        </td>
        <td class="data">
          <span class="byte">52</span>
        </td>
        <td class="label">
          
        </td>
        <td class="value">return undefined</td>
      </tr>
      
      <tr>
        <td class="address">
          <a class="address-text" id="address00B9" href="#address00B9">
            00B9
          </a>
        </td>
        <td class="data">
          
        </td>
        <td class="label">
          
        </td>
        <td class="value"></td>
      </tr>
    </tbody>
  </table>
      <style>body {
  font-family: sans-serif;
  margin: 20px;
}

table.visual-buffer {
  border-collapse: collapse;
  table-layout: fixed;
  width: 1000px;
  border-spacing: 0;
  font-size: 16px;
  font-family: 'Consolas', 'Courier New', Courier, monospace;
}

table.visual-buffer td {
  vertical-align: top;
  padding: 2px 0;
  margin: 0;
}

/* ---------------- Column 1: Address -------------------*/
table.visual-buffer col:nth-child(1) {
  width: 47px;
}

table.visual-buffer td:nth-child(1) {
  text-align: left;
  padding-right: 8px;
  border-top: 1px solid rgb(100,100,100);
}

.visual-buffer .address-text {
  font-size: 85%;
  /* font-weight: bold; */
  font-family: 'Consolas', 'Courier New', Courier, monospace;
  position: relative;
  top: -12px;
  border-radius: 2px;
  background-color: white;
  padding: 1px 4px;
  color: black;
  text-decoration: none;
}

.visual-buffer .address-text:hover {
  color: rgb(40, 40, 80);
  background-color: rgb(220, 220, 240);
}

/* ---------------- Column 2: Binary -------------------*/
table.visual-buffer col:nth-child(2) {
  background: rgb(40,40,50);
  width: 141px;
}

table.visual-buffer td:nth-child(2) {
  text-align: left;
  padding-left: 4px;
  padding-right: 4px;
  border-top: 1px solid rgba(255, 255, 255, 0.06);
  font-size: 0;
  height: 0px;
  overflow: hidden;
}

.visual-buffer .data .byte {
  display: inline-block;
  box-sizing: border-box;
  width: 29px;
  height: 29px;
  font-family: 'Consolas', 'Courier New', Courier, monospace;
  font-size: 17px;
  color: white;
  background: rgb(60,60,75);
  border-radius: 2px;
  text-align: center;
  padding-top: 6px;
  margin: 2px 2px;
  overflow: hidden;
}

.visual-buffer .data .byte.pending {
  box-shadow: 0px 0px 5px rgba(255, 255, 255, 0.3) inset;
  border-radius: 3px;
}

/* ---------------- Column 3: Label -------------------*/
table.visual-buffer col:nth-child(3) {
  width: 300px;
}

table.visual-buffer td:nth-child(3) {
  text-align: right;
  padding-right: 5px;
  opacity: 0.7;
  border-top: 1px solid rgba(100, 100, 100, 0.25);
  padding-top: 10px;
}

/* ---------------- Column 4: Value -------------------*/
table.visual-buffer col:nth-child(4) {
  text-align: left;
}

table.visual-buffer td:nth-child(4) {
  border-top: 1px solid rgba(100, 100, 100, 0.25);
  padding-top: 10px;
}

.visual-buffer .pending-value {
  font-style: italic;
  opacity: 0.5;
}
</style>
      </body>
      </html>